packages/blade/src/components/IconButton/StyledIconButton.native.tsx

```diff
@@ -0,0 +1,55 @@
+import type { ReactElement } from 'react';
+import { useState } from 'react';
+import styled from 'styled-components/native';
+
+import type { StyledIconButtonProps } from './StyledIconButton.d';
+import { makeAccessible } from '~utils';
+import type { ColorContrastTypes } from '~tokens/theme/theme';
+
+type GetIconColorTokenParams = {
+  isPressed: boolean;
+  contrast: ColorContrastTypes;
+};
+
+type State = 'active' | 'default';
+type IconColorToken = `feedback.neutral.action.icon.link.${State}.${ColorContrastTypes}Contrast`;
+
+const getIconColorToken = ({ isPressed, contrast }: GetIconColorTokenParams): IconColorToken => {
+  const contrastType = contrast === 'high' ? 'highContrast' : 'lowContrast';
+  const state = isPressed ? 'active' : 'default';
+
+  return `feedback.neutral.action.icon.link.${state}.${contrastType}`;
+};
+
+type StyledPressableProps = {
+  contrast: ColorContrastTypes;
+};
+
+const StyledPressable = styled.Pressable<StyledPressableProps>({
+  alignSelf: 'flex-start', // ensure button only takes needed width
```

Comment:
I see, my rationale was if I place this button in a full width box it should appear to the left (start) and not automatically center.

Reply:
Yeah, that makes sense too. We can't predict what the consumer would want most of the times. I've been habitual to making it center because that has fit most use-cases in the past. Anyway, we've done center for Button, Link & Badge too. Would be better if we kept the same experience across components so that the consumer doesn't suddenly see some different behaviour for some components right?

Reply:
Resolved in https://github.com/razorpay/blade/pull/613/commits/a7c992a1205f00da61795e47039c46eb4457f64a